hibernate-core/src/main/java/org/hibernate/event/internal/EntityState.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.event.internal;
+
+import org.hibernate.engine.internal.ForeignKeys;
+import org.hibernate.engine.spi.EntityEntry;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.engine.spi.Status;
+import org.hibernate.internal.CoreLogging;
+import org.hibernate.internal.CoreMessageLogger;
+
+public enum EntityState {
+	PERSISTENT, TRANSIENT, DETACHED, DELETED;
+
+	static final CoreMessageLogger LOG = CoreLogging.messageLogger( EntityState.class );
+
+	/**
+	 * Determine whether the entity is persistent, detached, or transient
+	 *
+	 * @param entity The entity to check
+	 * @param entityName The name of the entity
+	 * @param entry The entity's entry in the persistence context
+	 * @param source The originating session.
+	 *
+	 * @return The state.
+	 */
+	public static EntityState getEntityState(
+			Object entity,
+			String entityName,
+			EntityEntry entry, //pass this as an argument only to avoid double looking
+			SessionImplementor source,
+			Boolean assumedUnsaved) {
+
+		if ( entry != null ) { // the object is persistent
+
+			//the entity is associated with the session, so check its status
```

Comment:
It all boils down to personal preference, but I would leave a space char after `//` to be consistent and avoid my eyes taking some used to it. Super minor off course.

Reply:
@NathanQingyangXu I agree with you :) but since Gavin is very focused on RX progress, I think we'll have to take this as is for now to not distract him, we can cleanup details later.